cmd/dinspect/main.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+	"os"
+	"runtime"
+	"strings"
+
+	"github.com/ikitiki/logical_backup/pkg/decoder"
+)
+
+var (
+	filePath string
+
+	Version  string
+	Revision string
+
+	GoVersion = runtime.Version()
+)
+
+func buildInfo() string {
+	return fmt.Sprintf("logical backup delta inspector tool %s git revision %s go version %s", Version, Revision, GoVersion)
+}
+
+func init() {
+	if len(os.Args) < 2 {
+		fmt.Fprintf(os.Stderr, "%s\n", buildInfo())
+		fmt.Fprintf(os.Stderr, "\nUsage:\n\t%s {deltafile}\n", os.Args[0])
+		os.Exit(1)
+	}
+
+	filePath = os.Args[1]
+	if _, err := os.Stat(filePath); os.IsNotExist(err) {
+		fmt.Fprintf(os.Stderr, "files %q does not exist", filePath)
+		os.Exit(1)
+	}
+}
+
+func main() {
+	fmt.Printf("reading %q segment file: \n\n", filePath)
+
+	fp, err := os.OpenFile(filePath, os.O_RDONLY, os.ModePerm)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "could not open file: %v\n", err)
+		os.Exit(1)
+	}
+	defer fp.Close()
+
+	lnBuf := make([]byte, 8)
```

Comment:
restore tool and inspect tool will use the same code in the future. as for now that part duplicates code from restore